mpc-core/src/protocols/plain.rs

```diff
@@ -125,10 +137,38 @@ impl<F: PrimeField> PrimeFieldMpcProtocol<F> for PlainDriver<F> {
         Ok(a.inverse().unwrap())
     }
 
+    fn inv_many(&mut self, a: &[Self::FieldShare]) -> std::io::Result<Vec<Self::FieldShare>> {
+        let mut res = Vec::with_capacity(a.len());
+
+        for a in a {
+            if a.is_zero() {
+                return Err(std::io::Error::new(
+                    std::io::ErrorKind::InvalidInput,
+                    "Cannot invert zero",
+                ));
+            }
+            res.push(a.inverse().unwrap());
+        }
+
+        Ok(res)
+    }
+
     fn neg(&mut self, a: &Self::FieldShare) -> Self::FieldShare {
         -*a
     }
 
+    fn neg_vec_in_place(&mut self, a: &mut Self::FieldShareVec) {
+        for x in a.iter_mut() {
+            *x = self.neg(x);
+        }
+    }
+
+    fn neg_vec_in_place_limit(&mut self, a: &mut Self::FieldShareVec, limit: usize) {
+        for x in a.iter_mut().take(limit) {
+            *x = self.neg(x);
+        }
+    }
+
```

Comment:
This is unfortunate... but we would need a slice type or the iterator stuff to be generic again to avoid this.